homeassistant/components/switch/arduino.py

```diff
@@ -28,24 +28,31 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     pins = config.get('pins')
     for pinnum, pin in pins.items():
         if pin.get('name'):
-            switches.append(ArduinoSwitch(pin.get('name'),
-                                          pinnum,
-                                          pin.get('type')))
+            switches.append(ArduinoSwitch(pinnum, pin))
     add_devices(switches)
 
 
 class ArduinoSwitch(SwitchDevice):
     """Representation of an Arduino switch."""
 
-    def __init__(self, name, pin, pin_type):
+    def __init__(self, pin, options):
         """Initialize the Pin."""
         self._pin = pin
-        self._name = name or DEVICE_DEFAULT_NAME
-        self.pin_type = pin_type
+        self._name = options.get('name') or DEVICE_DEFAULT_NAME
+        self.pin_type = options.get('type')
         self.direction = 'out'
-        self._state = False
+
+        default = options.get('default')
+        negate = options.get('negate')
+
+        self._state = default and default.lower() in ['yes', 'true']
+        is_negate = negate and negate.lower() in ['yes', 'true']
+
+        self.turn_on_handler = self._turn_off if is_negate else self._turn_on
```

Comment:
I'm not a big fan of this coding style. It is confusing to read. Could you rewrite this as:

```python
if is_negate:
    self.turn_on_handler = …
else:
    self.turn_on_handler = …
```

Instead of using `self._turn_on` as proxy methods, you could also set the handler directly to `arduino.BOARD.set_digital_out_high`